src/libtomahawk/resolvers/ScriptJob.cpp

```diff
@@ -90,9 +90,17 @@ ScriptJob::reportResults( const QVariantMap& data )
 
 
 void
+ScriptJob::reportResults( const QVariant& data )
+{
+    m_data_primitive = data;
+    emit done( data );
+}
+
+
+void
 ScriptJob::reportFailure( const QString& errorMessage )
 {
     emit error( errorMessage );
 
-    reportResults( QVariantMap() );
+    reportResultsMap( QVariantMap() );
```

Comment:
You should emit both here, also you made empty variant(map)s default parameters, so you don't need to pass them in here.